File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    s=list(s)
    stack=[s[0]]
    i=1
    while i!=len(s):
        if stack==[]:
            stack.append(s[i])
        elif s[i]==stack[-1]:
            stack.pop()
        else:
            stack.append(s[i])
        i+=1
    if stack==[]:
        return 1
    return 0

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s)
{
    let stack=[s[0]];
    let i=1
    while(i<s.length){
        if(stack===[]||stack[stack.length-1]!==s[i])
            stack.push(s[i]);
        else
            stack.pop();
        i+=1
    }
    return stack.length?0:1
}